hyperspy/tests/signal/test_eds_tem.py

```diff
@@ -327,3 +327,48 @@ def test_manual_remove_element(self):
         assert (
             sorted(s._xray_markers.keys()) ==
             ['Zn_Ka', 'Zn_La'])
+
+
+@lazifyTestClass
```

Comment:
It "works" in the sense that it does not crash for your small examples and gives a correct (non-lazy) answer.

It does not work as it's supposed to for lazy signals (which should be done both lazily and in blocks instead of just iterating dimensions and summing things). Instead it creates a numpy array to store the final and intermediate results, that essentially shrinks in every dimension as the algorithm iterates over them. So if you're not rebinning substantially in the first one (or maybe even up-scaling?) it "blows up" straight away.

Sorry for the short answer (I might go into detail at some point later this or next week if there is any need), but trust me - it does not do what it needs to for the lazy signals.